### ISSUE TITLE
Moving attribute to axis level keeps it editable at common level, blocking product edit save

### ISSUE THREAD

**Issue by stefandoorn** [akeneo/pim-community-dev#9893]
I'm using Akeneo v3.0.11.

Recently I moved an attribute from the `common` level to the `first variant axis` level (attribute is called `product_name`). It made more sense to define it at that level as it could be different. Anyway, for some of the products, if I go to edit them, I see the field still at common level. It's disabled. At the variant level I also see the field, not disabled. At the variant level I CAN change & save it. When I want to change something at the common level though, I get an error:

`"Cannot set the property "product_name" to this entity as it is not in the attribute set"`.

I checked all the family variants, and attribute `product_name` is placed at the first variant axis. 

I'm not really sure what is the issue here, also not really sure what is the solution.

**Comment by Doodoune** [comment]
Hi @stefandoorn,

When you move an attribute from a commun level to a variant axis, a job was add in the queue  `Compute family variant structure changes`. 
It's this job that "move" correctly attribute values concerned by this change from the common level to the variant axis level on your products & product models impacted.

Can you check if this job was correctly launched and fully complete?

Regards,

**Comment by stefandoorn** [comment]
I'm quite sure that task failed - probably due to out of memory - and therefore never finished. It would be great to have a retry function for background tasks. Also, due to the size of my product catalog I'm running regularly into memory issues (see also other issue) which might be interesting to dive into.

**Comment by MichelFalke** [comment]
I'm having the exact same problem as described here.

**Comment by mhelm75** [comment]
I'm experiencing this same issue on one of my Akeneo projects. In my case it seems that the compute_family_variant_structure_change job is failing because of product model codes that have non-alphanumeric value. For example:
`[2019-11-15 21:00:35] batch.ERROR: Encountered an error executing the step: Validation error for ProductModel with code "Apple_iPad_Pro_12.9_Liquid_Retina_Tablet" during family variant structure change [] []`
Note how the above code has "12.9" in it, which breaks the Akeneo identifier rules of alphanumeric and underscores only.

Another example:
`[2019-11-15 20:41:38] batch.ERROR: Encountered an error executing the step: Validation error for ProductModel with code "Flying _Mesh _Office_Chair" during family variant structure change [] []`
Note how the above code has spaces between Flying and _Mesh and spaces between _Mesh and _Office_Chair.

**Comment by SSMellow** [comment]
I have also experienced this bug. I'm using 2.3.36 version of Akeneo. Somehow, from the common level of product all attributes moved to Product level. I think that it should not be possible because of validator, but somehow it appeared here. Any suggestions?

**Comment by stefandoorn** [comment]
Unfortunately I'm still experiencing this issue, I'm not sure what makes the job fail but it doesn't seem to complete. 

@Doodoune Is there also a way to restart such a job? It's risky to do it manually from the database obviously.

**Comment by Doodoune** [comment]
Hi @stefandoorn,

Sorry for the delay.
The job "compute_product_model_descendant" is launched by a subscriber that listen to PostSave events on Product Models. So if you editing the product model, the job will be launched again.

Regards,

**Comment by bOnepain** [comment]
I have exactly the same problems on both sides (I mean when moving an attribute from the product model to a variant level, and the other way too). The tasklets aren't launched at all. 
I'm on PIM EE 3.2.31.
Now the whole database is messy. I don't know exactly which product models and products are affected. Right now I'm writing my own command to fix that (compare attributes configuration at family (variant) side and attribute on the product models / product and need to override the product saver so that the StorageEvents::PRE_SAVE event is not dispatch - otherwise raw_values are "reset" in a listener).

**Comment by stefandoorn** [comment]
Yes, the only issue there is that I'm not able to edit that one anymore, as it fails on a validation error (see above, "Cannot set the property "product_name" to this entity as it is not in the attribute set".). So I'm basically stuck on that product it seems. I could try it again, but I'm slightly afraid I'm going to mess up my database.

**Comment by stefandoorn** [comment]
I just tried it, and in case the family has too many products this runs out of memory easily. Then the behavior as above mentioned is happening, the product model cannot be edited anymore & the data is not available on the product level.

**Comment by stefandoorn** [comment]
Only way to get past this is removing these attributes from the `raw_values` in the database and making sure the products get these values again.

**Comment by stefandoorn** [comment]
Additional - it worked fine on a family that involved 30 products (30.000 products but not all had that attribute) for this attribute. But it failed on a family with approx >40.000 (180.000 products but not all had that attribute) products involved.

**Comment by SSMellow** [comment]
@stefandoorn maybe you managed this issue somehow? Because I face it too. At this moment I wrote a migration, which fixes all those products, but it's just workaround, not problem solving.

**Comment by peter-burger** [comment]
Anyone having this issue still? If have tried it with a complete demo installation of version 4 and just moved the MEDIA group in CODE: SHOES_SIZE from common to variant level axis one. Same thing...
